### PR TITLE
Debuggability in the new sensor world

### PR DESCRIPTION
**Pull request by mck1117**
- Hook up each sensor printing information about itself (next step is having the functional sensor print out info about the function in question)
- Rename `initSensors` to `initNewSensors` to prevent name collision
- Fix `efilib.h` - it didn't compile if it was the first/only thing included in a file, which somehow flew under the radar for a long time.
- Remove duplicate check from `printAnalogChannelInfo` - it already happens in `printAnalogChannelInfoExt`

### CODE REVIEW

**Comment by mck1117** [comment]
Here's what the output from `show_sensors` looks like:
```
2020-03-30_14_51_33_704: EngineState: Sensor "CLT" is not configured.
2020-03-30_14_51_33_704: EngineState: Sensor "IAT" is not configured.
2020-03-30_14_51_33_711: EngineState: Sensor "Oil Pressure" is not configured.
2020-03-30_14_51_33_854: EngineState: Sensor "TPS 1": Raw value: 0.98 Valid: 1 Converted value -8.03
2020-03-30_14_51_33_859: EngineState: Sensor "TPS 1 Primary" is not configured.
2020-03-30_14_51_33_862: EngineState: Sensor "TPS 1 Secondary" is not configured.
2020-03-30_14_51_33_865: EngineState: Sensor "TPS 2" is not configured.
2020-03-30_14_51_33_868: EngineState: Sensor "TPS 2 Primary" is not configured.
2020-03-30_14_51_33_873: EngineState: Sensor "TPS 2 Secondary" is not configured.
2020-03-30_14_51_33_875: EngineState: Sensor "Acc Pedal" is not configured.
2020-03-30_14_51_33_875: EngineState: Sensor "Acc Pedal Primary" is not configured.
2020-03-30_14_51_33_876: EngineState: Sensor "Acc Pedal Secondary" is not configured.
2020-03-30_14_51_33_876: EngineState: Sensor "Driver Acc Intent" proxied from sensor "TPS 1"
```

File: firmware/controllers/sensors/sensor.cpp [hunk 1]
```diff
@@ -12,6 +14,30 @@ struct SensorRegistryEntry {
 
 static SensorRegistryEntry s_sensorRegistry[static_cast<size_t>(SensorType::PlaceholderLast)] = {};
 
+static const char* s_sensorNames[] = {
+	"Invalid",
```
Comment (rusefi) [inline firmware/controllers/sensors/sensor.cpp]: dirty magic index matching? kind of hacky and poorly documentad coupling between enum and this array?

Comment (mck1117) [reply firmware/controllers/sensors/sensor.cpp]: that's what the static assert is for - it at least guarantees that the sizes match

File: firmware/util/efilib.h [hunk 2]
```diff
@@ -45,19 +45,6 @@ int indexOf(const char *string, char ch);
 float atoff(const char *string);
 int atoi(const char *string);
 
-#if defined(__cplusplus) && defined(__OPTIMIZE__)
-#include <type_traits>
```
Comment (rusefi) [inline firmware/util/efilib.h]: how is this related to "Debuggability in the new sensor world"? :)

Comment (mck1117) [reply firmware/util/efilib.h]: see second bullet point:
> Fix efilib.h - it didn't compile if it was the first/only thing included in a file, which somehow flew under the radar for a long time.

File: firmware/controllers/sensors/sensor.cpp (hunk 1 above)
Comment (rusefi) [reply firmware/controllers/sensors/sensor.cpp]: that's not enough. this has to designed better or documented better IMHO
